packages/truffle-box/lib/utils/unbox.js

```
@@ -1,90 +1,73 @@
-      }
-      accept();
-    });
-  });
+  const configURL = parseURL(
+    `${vcsurl(url)
+      .replace("github.com", "raw.githubusercontent.com")
+      .replace(/#.*/, "")}/master/truffle-box.json`
+  );
+
+  const options = {
+    method: "HEAD",
+    uri: `https://${configURL.host}${configURL.path}`,
+    resolveWithFullResponse: true,
+    simple: false
+  };
+
+  try {
+    const { statusCode } = await rp(options);
+    if (statusCode === 404) {
+      throw new Error(
+        `Truffle Box at URL ${url} doesn't exist. If you believe this is an error, please contact Truffle support.`
+      );
+    } else if (statusCode !== 200) {
+      throw new Error(
+        "Error connecting to github.com. Please check your internet connection and try again."
+      );
+    }
+  } catch (error) {
+    throw error;
+  }
 }
 
 function fetchRepository(url, dir) {
-  return new Promise(function(accept, reject) {
+  return new Promise((accept, reject) => {
     // Download the package from github.
     ghdownload(url, dir)
-      .on("err", function(err) {
+      .on("err", err => {
         reject(err);
       })
-      .on("end", function() {
+      .on("end", () => {
         accept();
       });
   });
 }
 
-function prepareToCopyFiles(tempDir, boxConfig) {
-  const needingRemoval = boxConfig.ignore || [];
+function prepareToCopyFiles(tempDir, { ignore }) {
+  const needingRemoval = ignore;
 
   // remove box config file
   needingRemoval.push("truffle-box.json");
   needingRemoval.push("truffle-init.json");
 
-  const promises = needingRemoval
+  needingRemoval
     .map(fileName => path.join(tempDir, fileName))
-    .map(
-      filePath =>
-        new Promise((resolve, reject) => {
-          fs.remove(filePath, error => {
-            if (error) return reject(error);
-            resolve();
-          });
-        })
-    );
-
-  return Promise.all(promises);
+    .forEach(filePath => {
+      try {
+        fse.removeSync(filePath);
+      } catch (error) {
```

Can we remove this catch?

---

I assumed `removeSync` would throw if a file needing removal wasn't there, but looks like `removeSync` will just silently do nothing.